README.md

```diff
@@ -1 +1,186 @@
+   - Enables broadcasting messages to all users, groups, or specific users
+   - [Create a Web PubSub resource](https://learn.microsoft.com/en-us/azure/azure-web-pubsub/howto-develop-create-instance)
+   - You'll need the connection string from this service
+
+3. **Cosmos DB** (Optional - referenced in Api.csproj)
+   - The project has Cosmos DB extensions installed but may not be actively used
+   - Can be used for message persistence if needed
+
+4. **Application Insights** (Optional but recommended)
+   - Configured in the API for telemetry and monitoring
+   - Connection string can be provided via configuration
+
+## User Secrets Configuration
+
+The API project uses .NET User Secrets for local development. You need to configure the following secrets:
+
+### Required Secrets
+
+Navigate to the `Api` directory and run:
+
+```bash
+cd Api
+dotnet user-secrets set "WebPubSubConnectionString" "<your-web-pubsub-connection-string>"
+```
+
+**To get your Web PubSub connection string:**
+1. Go to your Azure Web PubSub resource in the Azure Portal
+2. Navigate to "Keys" under Settings
+3. Copy the "Connection String"
+
+### Optional Secrets
+
+```bash
+# Optional: Custom hub name (defaults to "notifications" if not set)
+dotnet user-secrets set "WebPubSubHubName" "notifications"
+
+# Optional: Application Insights connection string
+dotnet user-secrets set "APPLICATIONINSIGHTS_CONNECTION_STRING" "<your-app-insights-connection-string>"
+```
+
+## Running the Application Locally
+
+### Prerequisites
+
+- [.NET 9.0 SDK](https://dotnet.microsoft.com/download/dotnet/9.0)
+- [Azure Functions Core Tools v4](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local)
+- [Azure Static Web Apps CLI](https://azure.github.io/static-web-apps-cli/)
+- An Azure Web PubSub resource (see Azure Resources section)
+
+### Installing the SWA CLI
+
+The Azure Static Web Apps CLI (SWA CLI) is required to run the application locally as it emulates the Azure Static Web Apps environment.
+
+Install globally using npm:
+
+```bash
+npm install -g @azure/static-web-apps-cli
+```
+
+Or using yarn:
+
+```bash
+yarn global add @azure/static-web-apps-cli
+```
+
+Verify installation:
+
+```bash
+swa --version
+```
+
+**Learn more:** [SWA CLI Documentation](https://azure.github.io/static-web-apps-cli/)
+
+### Running with SWA CLI
+
+The application requires both the API and Client to be running simultaneously. The SWA CLI orchestrates this.
+
+1. **Configure your secrets** (see User Secrets Configuration section above)
+
+2. **Start the API** (in one terminal):
+   ```bash
+   cd Api
+   func start
+   ```
+   
+   The API will start on port 7071.
+
+3. **Start the Client** (in another terminal):
+   ```bash
+   cd Client
+   dotnet run
+   ```
+   
+   The client will start on port 7001 (HTTP) or 7000 (HTTPS).
+
+4. **Start SWA CLI** (in a third terminal):
+   ```bash
+   swa start http://localhost:7001 --api-location http://localhost:7071
```

Comment:
The code is deliberately designed to run out of the box so this is not needed as 7071 is the default